clang/test/Driver/sycl-cuda-arch-macro.cpp

```diff
@@ -0,0 +1,3 @@
+// RUN: %clangxx %s -fsycl -nocudalib -fno-sycl-libspirv -fsycl-targets=nvptx64-nvidia-cuda -Xsycl-target-backend --offload-arch=sm_80 -E -dM \
+// RUN: | FileCheck --check-prefix=CHECK-CUDA-ARCH-MACRO %s
+// CHECK-CUDA-ARCH-MACRO-NOT: #define __CUDA_ARCH__ {{[0-9]+}}
```

Comment:
Given the original nature of the driver change, one thing we should also verify is that `-fcuda-is-device` is not supplied with `-fsycl -fsycl-targets=nvptx64-nvidia-cuda` for the device compilation (`-###` check)

Reply:
Thanks for being rigorous here! That's definitely what the original change prevented, where the unwanted macro was a product of this. I've added the test and updated the PR description accordingly.
Let me know if you think something needs to be updated.